### PR TITLE
Sesar indexing fixes

### PR DESCRIPTION
**Pull request by hyunssong**
Updates to make SESAR indexing to work. Created a new set of test records (as the previous one had wrong igsns), made sure that when extracting text fields from the record is done in a consistent manner, update codes to allow Top3 high confidence results are going to be returned.

### CODE REVIEW

File: isamples_metadata/taxonomy/OpenContextClassifierInput.py [hunk 1]
```diff
@@ -11,14 +11,15 @@ def __init__(self, thing):
         self.OC_material_field = "Consists of_label"
         self.OC_sample_field = "Has type_label"
 
-    def build_text(self, description_map, labelType):
+    def build_text(self, description_map, labelType, description_field):
         """Return the concatenated text of informative fields in the
         description_map"""
         concatenated_text = ""
-        for key, value in description_map.items():
+        for field in description_field:
```
Comment (hyunssong) [inline isamples_metadata/taxonomy/OpenContextClassifierInput.py:18]: this is going to be further updated in a separate branch where I will add updates for OpenContext.

File: isamples_metadata/taxonomy/SESARClassifierInput.py [hunk 2]
```diff
@@ -79,20 +80,24 @@ def __init__(self, thing):
...
-                # empty field
-                continue
-            elif labelType == "material" and key != self.SESAR_material_field:
-                concatenated_text += value + " , "
-            elif labelType == "sample" and key != self.SESAR_sample_field:
-                concatenated_text += value + " , "
+        # use the order of the fields to build text
+        # to have consistent text content
+        for field in field_order:
+            if field in description_map:
```
Comment (hyunssong) [inline isamples_metadata/taxonomy/SESARClassifierInput.py:89]: Instead of extracting text fields from the dictionary `description_map,` fix the order of the fields that are going to be extracted using `field_order` when building the text.

File: isamples_metadata/taxonomy/Model.py [hunk 3]
```diff
@@ -29,11 +29,14 @@ def predict(self, text):
...
-        predicted_class = predicted_class.cpu().item()
-        prob = logits.softmax(dim=-1)[0][predicted_class].item()
-        return (
-            self.config["CLASS_NAMES"][predicted_class],
-            prob
-        )
+        # convert logits into probability
+        prob  =  logits.softmax(dim=-1)[0]
+        # get the top 3 high confidence predictions
+        top_probs, indices = torch.topk(prob, 3)
```
Comment (hyunssong) [inline isamples_metadata/taxonomy/Model.py:35]: Display the top3 high confidence labels using `topk`. This extracts the highest 3 probable labels.

File: isamples_metadata/taxonomy/metadata_models.py [hunk 4]
```diff
@@ -211,11 +211,11 @@ def classify_by_machine(self, text: str) -> Tuple[str, float]:
...
         """
-        prediction, prob = self._model.predict(text)
-        return (
-            SESARClassifierInput.source_to_CV[prediction],
+        predictions = self._model.predict(text)
+        return [(
+            SESARClassifierInput.source_to_CV[label],
             prob
-        )
+        ) for (label, prob) in predictions]
```
Comment (hyunssong) [inline isamples_metadata/taxonomy/metadata_models.py:218]: updating to list structure as multiple tuples will be returned.

File: scripts/taxonomy/sesar_indexing_test_records.json [hunk 5]
```diff
@@ -0,0 +1,3166 @@
+{
```
Comment (dannymandel) [inline scripts/taxonomy/sesar_indexing_test_records.json]: Could you delete this file and replace the lines in `isamples_inabox/integration_tests/test_model_values.json` please?

File: isamples_metadata/SESARTransformer.py [hunk 6]
```diff
@@ -309,7 +309,7 @@ def has_material_categories(self) -> typing.List[str]:
             sesar_model = MetadataModelLoader.get_sesar_material_model()
             # load the model predictor
             smp = SESARMaterialPredictor(sesar_model)
-            categories = [smp.predict_material_type(self.source_record).value]
+            categories = [prediction.value for prediction in smp.predict_material_type(self.source_record)]
```
Comment (dannymandel) [inline isamples_metadata/SESARTransformer.py:312]: Do we expect these to actually return multiple useful values for SESAR?  Or do we still expect one?

Comment (hyunssong) [reply isamples_metadata/SESARTransformer.py:312]: I expect this will return multiple labels (top 3 prediction labels).

Comment (hyunssong) [reply isamples_metadata/SESARTransformer.py:312]: Thinking again, I think there might be an issue here. The test record currently contain the list of high confidence labels, and the confidence of each of such prediction. But the `has_material_category` only returns a list of labels. I think we might need an additional field to store confidence scores in future, if we want to compare those confidence values.

Comment (dannymandel) [reply isamples_metadata/SESARTransformer.py:312]: Yeah, for now we are ignoring the confidence values in the test because we don't have them stored anywhere.

Comment (dannymandel) [reply isamples_metadata/SESARTransformer.py:312]: (And confirmed that the tests pass with the multiple values)

**Review by dannymandel — APPROVED** [review]
Looks good and I verified that the tests are passing now, as well.  Thanks @hyunssong!
